srtcore/core.cpp

```diff
@@ -12515,602 +12559,599 @@ int CUDTGroup::recv(char* buf, int len, SRT_MSGCTRL& w_mc)
 
     size_t output_size = 0;
 
-RETRY_READING:
-
+    for (;;)
     {
         if (!m_bOpened || !m_bConnected)
         {
             LOGC(dlog.Error, log << boolalpha << "group/recv: ERROR opened=" << m_bOpened << " connected=" << m_bConnected);
             throw CUDTException(MJ_CONNECTION, MN_NOCONN, 0);
         }
```

Comment:
These changes are not related to the balancing implementation and look more like a refactoring, that was requested in #1134. Having these changes in this PR complicate the reviewing process.
- [ ] Move to separate PR